### ISSUE TITLE
Infinite loop / infinite recursion 

### ISSUE THREAD

**Issue by ahasall** [FabioAntunes/fish-nvm#32]
Hello,

First of all thank you for this package. It has been useful to me.

Since fish 3.0, I am encountering an issue where fish is sourced indefinitely.
This happen when I run commands that are added to `/usr/local/bin/` by `nvm_alias_command`.

It seems like the problem is related to `__nvm_run`. Also, when I comment the `__nvm_run` command in `/usr/local/bin/node` for instance, a new fish shell is launched but doesn't cause recursion

I have installed fish-nvm using Oh my fish.

Thanks,
Amadou

**Comment by FabioAntunes** [comment]
I haven't tested this on fish 3.0, thanks for submitting this I will have a look as soon as possible.

**Comment by ahasall** [comment]
Thank you very much. If there is anything I can do to assist you, please let me know.

**Comment by FabioAntunes** [comment]
I was not able to reproduce the loop problem, but I did found a bug on the `nvm_alias_command` because `omf` doesn't symlink files, running `nvm_alias_command` without any args would create commands with the current content of the folder where the function was running.

![screenshot 2019-02-16 at 14 52 59](https://user-images.githubusercontent.com/2544673/52901411-07449c00-31fb-11e9-82fb-67298a179158.png)


Could you try removing all the commands created on `/usr/local/bin/` by `nvm_alias_command` do a `omf update fish-nvm` and then run `nvm_alias_command`

**Comment by ahasall** [comment]
I never ran `nvm_alias_command` without argument.

I have also found that I had duplicate functions after  running the `nvm_alias_command`:
For example I had `npm.fish` in `/usr/local/bin` and the same in `$OMF_PATH/pkg/fish-nvm`.

For now, I have uninstalled `fish-nvm` and use `edc/bass` because I have a talk to give on February 21th. I will come back to you after that date.

Maybe you can also share with me your Docker image, so that I can try to reproduce the issue without messing up with my computer.

**Comment by FabioAntunes** [comment]
This is my Dockerfile

```
FROM ubuntu:18.10
RUN apt-get update && \
  apt-get install -y software-properties-common && \
  apt-add-repository -y ppa:fish-shell/release-3  && \
  apt-get update && \
  apt-get install -y fish && \
  apt-get install -y git && \
  apt-get install -y curl && \
  apt-get install -y python2.7 python-pip && \
  apt-get install -y locales && \
  locale-gen "en_US.UTF-8"
RUN git clone https://github.com/creationix/nvm.git ~/.nvm && \
  curl -L https://get.oh-my.fish > install && \
  fish install --path=~/.local/share/omf --config=~/.config/omf --noninteractive && \
  fish -c "omf install bass & omf install https://github.com/FabioAntunes/fish-nvm" && \
  fish -c "nvm install v10.15.1"

CMD /usr/bin/fish
```

Not sure if you are familiar with docker, but basically you just need to run this on the same folder where the Dockerfile is:

```
docker build . -t fish-dockerimage
```

Run a detached docker container

```
docker run -it fish-dockerimage fish
```

Run 

```
docker ps
```

You should be getting a container id, then ssh into that container, replace `{CONTAINER_ID}` with your container id. 

```
docker exec -it {CONTAINER_ID} fish
```

have fun

**Comment by jorgebucaran** [comment]
@FabioAntunes Does this problem also happen with [Fisher](https://github.com/jorgebucaran/fisher)? 

/cc @ahasall

**Comment by ahasall** [comment]
@jorgebucaran, I didn't try it with Fisher.

**Comment by jorgebucaran** [comment]
@ahasall Well, good luck with that!

**Comment by ahasall** [comment]
Thank you @jorgebucaran, is Fisher compatible with Fish 3 ?

**Comment by jorgebucaran** [comment]
Sure, but I'm curious, what made you wonder it wasn't?

**Comment by jorgebucaran** [comment]
@ahasall Yes, it is, but I'm curious, what made you wonder it wasn't?

**Comment by ahasall** [comment]
Because Fish 3 was released not so long time ago. I also see on Fisher's Readme that it supports Oh My Fish packages. I will have a look into Fisher. Currently I am using Oh My Fish.

**Comment by FabioAntunes** [comment]
> @FabioAntunes Does this problem also happen with [Fisher](https://github.com/jorgebucaran/fisher)?

Nah, at leat I have been using fisher and fish-nvm and never had an issue.

**Comment by FabioAntunes** [comment]
But @ahasall before you decide to move to fisher, help me debug this on omf, so we can help future users.

**Comment by ahasall** [comment]
@FabioAntunes, no problem.
I will try to reproduce this issue on my Linux Mint. (I am using macOS for now)

**Comment by ahasall** [comment]
I have been able to reproduce the problem inside a Docker container.
```bash
rm -rf $OMF_CONFIG
git clone https://github.com/ahasall/omf-config.git $OMF_CONFIG

rm -rf .config/fish
git clone https://github.com/ahasall/fish-config.git .config/fish

nvm_alias_command

fish
yarn
```

**Comment by FabioAntunes** [comment]
Fixed by #33 just update fish-nvm. thanks for reporting this @ahasall
